### PR TITLE
feat: add temporal reasoning test suite (16 tests, Tier 1)

### PR DESCRIPTION
**Pull request by tkarcheski**
## Summary

Adds a new `temporal-reasoning` test suite that tests three distinct LLM temporal reasoning capabilities. Temporal reasoning is a critical skill for emerging LLM use-cases such as scheduling assistants, calendar tools, deadline trackers, and historical-analysis agents — and it was the only major reasoning category not yet represented in the harness.

All 16 tests use Tier 1 / `verify:python` grading: responses are verified deterministically via structured output extraction, with no LLM judge required.

## How to review

**Start here:** Read `src/rfc/temporal_reasoning_keywords.py` — the three keyword methods and two private extraction helpers are the core of this change.

**Key changes:**
1. `src/rfc/temporal_reasoning_keywords.py` — new keyword library with `Evaluate Sequence Order`, `Evaluate Duration`, `Evaluate Temporal Word Problem`; private `_extract_letter_sequence` and `_extract_integer` helpers (critical — review extraction logic carefully)
2. `tests/test_temporal_reasoning_keywords.py` — 42 pytest unit tests covering helpers and all three keywords, including edge cases (prose responses, deduplication, case sensitivity, empty responses)
3. `robot/temporal_reasoning/` — 3 test files, 1 resource file, 3 YAML variable files, 1 `__init__.robot`
4. `config/test_suites.yaml` + `config/local_models.yaml` — suite registration (mechanical, 5 lines each)

**What to ignore:** The YAML variable files are data — each entry has a `rationale` field confirming the arithmetic. The resource file is boilerplate. Focus review time on the Python keyword library (step 1) and the extraction helpers.

## Evidence of testing

<details>
<summary>pytest output</summary>

```
2509 passed, 1 skipped, 6 warnings in 10.03s
(+42 new tests from test_temporal_reasoning_keywords.py, all pass)
```
</details>

<details>
<summary>pre-commit output</summary>

```
check yaml...............................................................Passed
check json...............................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
check for merge conflicts................................................Passed
ruff (legacy alias)......................................................Passed
ruff format..............................................................Passed
mypy.....................................................................Passed
Block FTP usage..........................................................Passed
```
</details>

<details>
<summary>code-quality-check output</summary>

```
All checks passed!
src/rfc/temporal_reasoning_keywords.py    78      0   100%
TOTAL                                   9096   1123    88%
Required test coverage of 50.0% reached. Total coverage: 87.65%
2509 passed, 1 skipped, 255 warnings in 18.06s
```
</details>

<details>
<summary>robot-dryrun output</summary>

```
603 tests, 591 passed, 12 failed
(up from 587/575/12 — 16 new temporal reasoning tests all pass;
 same 12 pre-existing Playwright/Browser failures unrelated to this change)
```
</details>

## Critical changes

None. No public API changes, no database schema changes, no config format changes. New suite is fully additive — the existing test harness is unchanged.

## Checklist

- [x] All pytest tests pass
- [x] pre-commit passes
- [x] code-quality-check passes (ruff + mypy)
- [x] robot-dryrun passes
- [x] Self-reviewed diff — no scope creep, no debug prints
- [x] Changes comply with `ai/agents.md` contract
- [x] Changes comply with `ai/testing.md` tier rules
- [x] New Robot tests have `tier:*` and `verify:*` tags
- [x] No unresolved `TODO`/`FIXME` in changed files
- [x] Type hints on all new Python code
- [x] Commit history is atomic and bisectable
- [x] Version bump confirmed (1.10.2 → 1.10.3, patch)

https://claude.ai/code/session_01Y5TcYtyvgwjPFd5ChhGJ14

---
_Generated by [Claude Code](https://claude.ai/code/session_01Y5TcYtyvgwjPFd5ChhGJ14)_

### CODE REVIEW

File: src/rfc/temporal_reasoning_keywords.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+# letters embedded inside words such as "Answer" or "Before".
+_LETTER_RE = re.compile(r"(?<![A-Za-z])([A-Ea-e])(?![A-Za-z])")
+
+# Matches the first integer in a string (no leading sign required).
+_INTEGER_RE = re.compile(r"\b(\d+)\b")
+
+# Matches a negation word immediately before the cursor position (fixed-width
+# alternatives so Python's lookbehind accepts them).
+_NEGATION_BEFORE_RE = re.compile(
+    r"\b(?:not|no|never|isn't|wasn't|don't|doesn't)\s*$",
```
[the remark this answers is not in the document]
Comment (tkarcheski) [reply src/rfc/temporal_reasoning_keywords.py:61]: Fixed in commit `cce15bf`.

The root cause turned out to be a character encoding trap: `str.maketrans` values written as string literals (e.g. `"'"`) were silently saved as U+2019 smart apostrophes by the editor, so the translate table was mapping typographic quotes to *themselves* rather than to U+0027. As a result `_NEGATION_BEFORE_RE` — which uses ASCII apostrophes — never matched after normalisation.

Fixed by using explicit hex ordinals as values:

```python
_SMART_APOSTROPHES = str.maketrans({
    0x2018: 0x0027,  # LEFT single quotation mark  '
    0x2019: 0x0027,  # RIGHT single quotation mark '
    0x02BC: 0x0027,  # modifier letter apostrophe  ʼ
    0x0060: 0x0027,  # grave accent                `
    0x00B4: 0x0027,  # acute accent                ´
})
```

Integer ordinals are unambiguous regardless of file encoding. Tests now cover both U+2018 and U+2019 variants (`test_smart_quote_isnt_fails`, `test_smart_quote_wasnt_fails`), and all 63 unit tests pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
